pants

```diff
@@ -2,6 +2,39 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+REPO_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd -P)
+
+# TODO(#7152): remove this after the python 3 migration!
+function clean_if_interpreter_changed() {
+  _python_version_to_run="$1"
+
+  if [[ -n "${_recursive_pants_shell_script_invocation:-}" ]]; then
+    return 0
+  fi
+
+  _python_version_file="${REPO_ROOT}/.pants-python-version"
+
+  if [[ -f "$_python_version_file" ]]; then
+    _prev_run_python_version="$(cat "$_python_version_file")"
+    if [[ "$_python_version_to_run" == "$_prev_run_python_version" ]]; then
+      return 0
+    fi
+  else
+    cat >&2 <<EOF
+${_python_version_file} not found. Forcing an initial clean-all...
```

Comment:
Tried this out locally and found the full path being printed a little jarring. I think this would be a nicer error message if it said something like:

```
.pants-python-version not found in the Pants root. Initiating an initial clean-all and creating this file...
```

Reply:
Will do!

Reply:
Done!